src/firstboot/firstboot.c

```diff
@@ -536,6 +536,8 @@ static int prompt_root_password(void) {
                 r = ask_password_tty(-1, msg1, NULL, 0, 0, NULL, &a);
                 if (r < 0)
                         return log_error_errno(r, "Failed to query root password: %m");
+                if (strv_length(a) > 1)
+                        log_warning("Received multiple passwords.");
```

Comment:
i think this should fail, because it is really unexpected. i.e. 

```c
if (strv_length(a) != 1) {
        log_warning("Received multiple passwords, where we expected one.");
        return -EINVAL;
}
```

or so